llama-index-integrations/vector_stores/llama-index-vector-stores-oracledb/llama_index/vector_stores/oracledb/base.py

```diff
@@ -0,0 +1,656 @@
+
+    # Combine all parts
+    ddl_assembly = (
+        base_sql + accuracy_part + distance_part + parameters_part + parallel_part
+    )
+    # Format the SQL with values from the params dictionary
+    ddl = ddl_assembly.format(**config)
+
+    # Check if the index exists
+    if not _index_exists(client, config["idx_name"]):
+        with client.cursor() as cursor:
+            cursor.execute(ddl)
+            logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+def _create_ivf_index(
+    client: Connection,
+    table_name: str,
+    distance_strategy: DistanceStrategy,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    # Default configuration
+    defaults = {
+        "idx_name": "IVF",
+        "idx_type": "IVF",
+        "neighbor_part": 32,
+        "accuracy": 90,
+        "parallel": 8,
+    }
+
+    config = _create_config(defaults, params)
+
+    # Base SQL statement
+    idx_name = config["idx_name"]
+    base_sql = f"CREATE VECTOR INDEX {idx_name} ON {table_name}(embedding) ORGANIZATION NEIGHBOR PARTITIONS"
+
+    # Optional parts depending on parameters
+    accuracy_part = " WITH TARGET ACCURACY {accuracy}" if "accuracy" in config else ""
+    distance_part = f" DISTANCE {_get_distance_function(distance_strategy)}"
+
+    parameters_part = ""
+    if "idx_type" in config and "neighbor_part" in config:
+        parameters_part = f" PARAMETERS (type {config['idx_type']}, neighbor partitions {config['neighbor_part']})"
+
+    # Always included part for parallel
+    parallel_part = f" PARALLEL {config['parallel']}"
+
+    # Combine all parts
+    ddl_assembly = (
+        base_sql + accuracy_part + distance_part + parameters_part + parallel_part
+    )
+    # Format the SQL with values from the params dictionary
+    ddl = ddl_assembly.format(**config)
+
+    # Check if the index exists
+    if not _index_exists(client, config["idx_name"]):
+        with client.cursor() as cursor:
+            cursor.execute(ddl)
+        logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+def drop_table_purge(client: Connection, table_name: str):
+    if _table_exists(client, table_name):
+        cursor = client.cursor()
+        with cursor:
+            ddl = f"DROP TABLE {table_name} PURGE"
+            cursor.execute(ddl)
+        logger.info("Table dropped successfully...")
+    else:
+        logger.info("Table not found...")
+
+
+@_handle_exceptions
+def drop_index_if_exists(client: Connection, index_name: str):
+    if _index_exists(client, index_name):
+        drop_query = f"DROP INDEX {index_name}"
+        with client.cursor() as cursor:
+            cursor.execute(drop_query)
+            logger.info(f"Index {index_name} has been dropped.")
+    else:
+        logger.exception(f"Index {index_name} does not exist.")
+
+
+class OraLlamaVS(VectorStore):
```

Comment:
This should extend BasePydanticVectorStore

Reply:
Please check out any other vector store for the pattern

Reply:
This has been fixed by shailendra